### ISSUE TITLE
Fl_Shared_Image won't load a SVG image with a BOM

### ISSUE THREAD

**Issue by fire-eggs** [fltk/fltk#247]
fl_check_images() in fl_images_core.cxx fails to recognize an SVG file which has a BOM.

These lines will fail:
```
  if ((count >= 5 && memcmp(buf, "<?xml", 5) == 0) ||
      (count >= 4 && memcmp(buf, "<svg", 4) == 0))
    return new Fl_SVG_Image(name);
```
because with a BOM the first 3 bytes in `buf` are the BOM and not the target strings.

The attached ZIP contains two files which are identical except for the presence of a BOM for one. FLTK will load the file w/o a BOM and will fail to load the file with a BOM.

[samples.zip](https://github.com/fltk/fltk/files/6757490/samples.zip)

**Comment by ManoloFLTK** [comment]
@fire-eggs: could you, please, check this issue is now solved in the git repository?

**Comment by Albrecht-S** [comment]
@ManoloFLTK Unfortunately your fix is not entirely correct (missing length check) and it's IMHO unnecessary to convert the BOM to UTF-8 since we know the UTF-8 BOM sequence anyway.

Since I've been working on a fix in parallel and was about to commit it, when I saw your commit, I decided to commit my mods nevertheless. Sorry for the double work...

Side note: I made a minimal change for further simplification (of the existing length check): instead of checking only `"<svg"` I compare now `"<svg "` (trailing space) which shouldn't have any impact on the behavior.

Note: tested with both provided images (samples.zip) and both images gzip'ed as well.

**Comment by wcout** [comment]
Checking for "`<svg `" will rule out e.g. "`<svg\t"` which is a valid notation.

**Comment by fire-eggs** [comment]
Other than @wcout 's point, it looks good.

**Comment by Albrecht-S** [comment]
@wcout Thanks for the heads-up, fixed in 2e9d13c1a23e43eab9ed70a12e06245405d11266.

@fire-eggs Thanks for confirmation. Looks like we can close this issue. Any comments, objections?

**Comment by fire-eggs** [comment]
Not from me. Thank you!

**Comment by ManoloFLTK** [comment]
The check for presence of BOM is now done twice.
Also, I disagree a length check was missing.

**Comment by Albrecht-S** [comment]
> The check for presence of BOM is now done twice.

@ManoloFLTK Thanks for finding this, I meant to overwrite your changes and made a mistake when merging your commit and mine. Fixed in 1008cdfab27609a6f6a0e82dadad9fd9cbd8a66d (hopefully the last update).

> Also, I disagree a length check was missing.

We had issues with accessing uninitialized data. I admit that it's a theoretical case; if you create a file with less than three bytes and gzip this file. Without the length check you would access three bytes of the variable `header2` that was written to with less than three bytes. Hypothetical but possible, here's my proof (with line numbers for comments below):
```
  1 $ echo "" | awk '{printf("%c%c", "\xef", "\xbb")}' | gzip -c > bom2.svgz
  2 $ gunzip -d < bom2.svgz | xxd -g1
  3 00000000: ef bb                                            ..
  4 $ bin/test/pixmap_browser
  5 buf = 0x7ffe9af5a1e0, count = 2
  6 buf[1-3] = ef, bb, bf
  7 buf = 0x7ffe9af5a1e3, count = -1
```

1. create 'bom2.svgz' which is a gzip'ed version of the first two bytes of the bom (there may be an easier way to do this, but it works)
2. unzip and dump with xxd to verify contents of bom2.svgz
3. output
4. execute pixmap_browser
5. output buffer address (`header2`) and count after gzip'ing
6. first three bytes of buffer [1]
7. output buffer address and count after bom check, count = -1

[1] I initialized the buffer `header2` with a copy of the bom to make this test match the bom (uninitialized data can have any contents). Otherwise the check would have failed, but accessing the unitialized data would have happened anyway.
